src/nomad_catalysis/schema_packages/catalysis.py

```diff
@@ -0,0 +1,378 @@
+    archive.
+    '''
+    if not archive.results:
+        archive.results = Results()
+    if not archive.results.properties:
+        archive.results.properties = Properties()
+    if not archive.results.properties.catalytic:
+        archive.results.properties.catalytic = CatalyticProperties()
+    if not archive.results.properties.catalytic.catalyst:
+        archive.results.properties.catalytic.catalyst = Catalyst()
+
+
+def get_nested_attr(obj, attr_path):
+    '''helper function to retrieve nested attributes'''
+    for attr in attr_path.split('.'):
+        obj = getattr(obj, attr, None)
+        if obj is None:
+            return None
+    return obj
+
+
+def set_nested_attr(obj, attr_path, value):
+    '''helper function to set nested attributes'''
+    for attr in attr_path.split('.'):
+        obj = getattr(obj, attr, None)
+        if obj is None:
+            return
+    setattr(obj, attr[-1], value)
+
+
+class Preparation(ArchiveSection):
+    preparation_method = Quantity(
+        type=str,
+        shape=[],
+        description="""
+          Classification of the dominant preparation step
+          in the catalyst synthesis procedure.
+          """,
+        a_eln=dict(
+            component='EnumEditQuantity',
+            props=dict(
+                suggestions=[
+                    'precipitation',
+                    'hydrothermal',
+                    'flame spray pyrolysis',
+                    'impregnation',
+                    'calcination',
+                    'unknown',
+                ]
+            ),
+            links=['https://w3id.org/nfdi4cat/voc4cat_0007016'],
+        ),
+    )
+
+    preparation_entry_reference = Quantity(
+        type=Process,
+        shape=[],
+        description="""
+        A reference to the entry that contains the details of the preparation method.
+        """,
+        a_eln=dict(component='ReferenceEditQuantity'),
+    )
+
+    preparator = Quantity(
+        type=str,
+        shape=[],
+        description="""
+        The person or persons preparing the sample in the lab.
+        """,
+        a_eln=dict(component='EnumEditQuantity'),
+    )
+
+    preparing_institution = Quantity(
+        type=str,
+        shape=[],
+        description="""
+        institution at which the sample was prepared
+        """,
+        a_eln=dict(
+            component='EnumEditQuantity',
+            props=dict(
+                suggestions=[
+                    'Fritz-Haber-Institut Berlin / Abteilung AC',
+                    'Fritz-Haber-Institut Berlin / ISC',
+                ]
+            ),
+        ),
+    )
+
+    def normalize(self, archive, logger):
+        super().normalize(archive, logger)
+
+
+class SurfaceArea(ArchiveSection):
+    m_def = Section(
+        label_quantity='method_surface_area_determination',
+        a_eln=ELNAnnotation(label='Surface Area'),
+    )
+
+    surface_area = Quantity(
```

Comment:
Description specifying the type of surface area

Reply:
done